### PR TITLE
ardour: give it all the libs it asks for. ardour3: bugfix

### PR DESCRIPTION
**Pull request by magnetophon**
The latest ardour3 release loses audio and midi
files under some conditions.
https://github.com/Ardour/ardour/commit/7024232855d268633760674d34c096ce447b7240
fixes that.

### CODE REVIEW

File: pkgs/applications/audio/ardour/ardour3.nix [hunk 1]
```diff
@@ -47,17 +53,17 @@ stdenv.mkDerivation rec {
...
 
-  configurePhase = "python waf configure --optimize --prefix=$out";
+  configurePhase = "python waf configure --optimize --docs --with-backends=jack,alsa --prefix=$out";
 
   buildPhase = "python waf";
 
   # For the custom ardour clearlooks gtk-engine to work, it must be
   # moved to a directory called "engines" and added to GTK_PATH
+    #mkdir -pv $out/gtk2/engines
+    #cp build/libs/clearlooks-newer/libclearlooks.so $out/gtk2/engines/
```
Comment (jagajaga) [inline pkgs/applications/audio/ardour/ardour3.nix]: Maybe just delete, not comment?

File: pkgs/applications/audio/ardour/default.nix [hunk 2]
```diff
@@ -50,17 +47,17 @@ stdenv.mkDerivation rec {
...
 
-  configurePhase = "python waf configure --with-backend=alsa,jack --optimize --prefix=$out";
+  configurePhase = "python waf configure --optimize --docs --with-backends=jack,alsa --prefix=$out";
 
   buildPhase = "python waf";
 
   # For the custom ardour clearlooks gtk-engine to work, it must be
   # moved to a directory called "engines" and added to GTK_PATH
+    #mkdir -pv $out/gtk2/engines
+    #cp build/libs/clearlooks-newer/libclearlooks.so $out/gtk2/engines/
```
Comment (jagajaga) [inline pkgs/applications/audio/ardour/default.nix]: Same question here.

**Comment by jagajaga** [comment]
Btw, thank you for your audio pkgs work!

**Comment by magnetophon** [comment]
@jagajaga thanks for the feedback :)

File: pkgs/applications/audio/ardour/ardour3.nix [hunk 3]
```diff
@@ -47,17 +53,12 @@ stdenv.mkDerivation rec {
     sed -e 's|^#!/usr/bin/env.*$|#!${perl}/bin/perl|g' -i tools/*.pl
```
Comment (codyopel) [inline pkgs/applications/audio/ardour/ardour3.nix]: I would recommend using `patchShebangs`.
```nix
{
   patchPhase = ''
     # Entire directory
     patchShebangs ./tools/
     # or a single file
     patchShebangs ./<path>/<file>
   '';
}
```

Comment (codyopel) [inline pkgs/applications/audio/ardour/ardour3.nix]: Also a note that when you use `patchShebangs` you need to include the interpreter as a `nativeBuildInput` or `buildInput`

Comment (jagajaga) [inline pkgs/applications/audio/ardour/ardour3.nix]: Good idea.

**Comment by codyopel** [comment]
@magnetophon @cillianderoiste 
Out of curiosity is ardour3 still necessary?

**Comment by magnetophon** [comment]
@codyopel the devs say it isn't, but I like to have a fallback option.

**Comment by cillianderoiste** [comment]
This builds locally, I'll merge.

**Pull request merged**